tests/autoscaler/pool_manager_test.py

```diff
@@ -338,6 +340,15 @@ def test_killable_nodes_under_target_capacity(self, mock_logger, mock_pool_manag
         assert mock_pool_manager._choose_nodes_to_prune(125, None) == {}
         assert "would take us under our target_capacity" in mock_logger.info.call_args[0][0]
 
+    def test_killable_node_cordoned(self, mock_logger, mock_pool_manager):
+        mock_pool_manager.get_node_metadatas = mock.Mock(
+            return_value=[_make_metadata("sfr-1", "i-1", is_cordoned=True)]
+        )
+
+        mock_pool_manager.max_tasks_to_kill = 100
+
+        assert mock_pool_manager._choose_nodes_to_prune(100, None) == {}
```

Comment:
It seems like here you are testing the return from an internal function , If possible we should avoid it. maybe you can try mocking the draining_client here https://sourcegraph.com/github.com/Yelp/clusterman@517097a48fd2430589135d2d3b4c3a7b312353d1/-/blob/clusterman/autoscaler/pool_manager.py?L181 and assert that this client was never called ( which will confirm that we got an empty map for the nodes

Reply:
I would agree if all the other tests in the class didn't do the same thing .-. I would want this specific test to be out of place, so I need to either fix all of them, or keep it that way.

Do you think it's worth updating all of them?

There's also the fact that there's some contention over whether testing private methods in unit tests is acceptable. It definitely isn't in integration tests, but we might need a team discussion if we'll be avoiding testing private methods.

Reply:
yeah , I agree testing internal functions are already part of the codebase so let's continue it for now.